### PR TITLE
[10.6] [PR 3655] refactor ci to work with all branches

### PR DESCRIPTION
**Pull request by mmattel**
Backport of PR #3655

### CODE REVIEW

File: .drone.star [hunk 1]
```diff
@@ -4,7 +4,8 @@ def main(ctx):
     # Version shown as latest in generated documentations
     # It's fine that this is out of date in version branches, usually just needs
     # adjustment in master/deployment_branch when a new version is added to site.yml
-    latest_version = "10.6"
+    latest_version = "10.7"
```
Comment (xoxys) [inline .drone.star:7]: Wrong version for 10.6

**Review by xoxys — CHANGES_REQUESTED** [review]
The version branches are still using outdated syntax e.g. https://github.com/owncloud/docs/blob/10.6/.drone.star#L75 while this is fixed already in the master. How do we get these changes into the version branches? Same for https://github.com/owncloud/docs/blob/10.6/.drone.star#L57

File: .drone.star [hunk 2]
```diff
@@ -13,14 +14,53 @@ def main(ctx):
 
     # Version branches never deploy themselves, but instead trigger a deployment in deployment_branch
     # This must not be changed in version branches
-    deployment_branch = "master"
+    deployment_branch = default_branch
+    pdf_branch = default_branch
```
Comment (xoxys) [inline .drone.star:18]: As described in the master branch PR, `pdf_branch` need to be changed on version branches to `pdf_branch = base_branch`

File: .drone.star (hunk 1 above)
Comment (mmattel) [inline .drone.star:7]:
```suggestion
    latest_version = "10.6"
```

File: .drone.star (hunk 2 above)
Comment (mmattel) [inline .drone.star:18]:
```suggestion
    pdf_branch = base_branch
```

**Comment by xoxys** [comment]
I've added a commit to fix the antora syntax as well.

**Review by xoxys — APPROVED** [review]

**Pull request merged**